WordPress/src/main/java/org/wordpress/android/ui/domains/management/DomainStatusRow.kt

```diff
@@ -0,0 +1,90 @@
+package org.wordpress.android.ui.domains.management
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+@Composable
+fun StatusRow(
+    status: DomainStatus,
+    expiry: LocalDate,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier,
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Box(
+            modifier = Modifier
+                .size(8.dp)
+                .clip(CircleShape)
+                .background(status.statusUiState.indicatorColor),
+        )
+        Spacer(modifier = Modifier.width(8.dp))
+        Text(
+            text = status.statusUiState.text,
+            color = status.statusUiState.textColor,
+            style = if (status.statusUiState.isBold) {
+                LocalTextStyle.current.copy(fontWeight = FontWeight.Bold)
+            } else {
+                LocalTextStyle.current
+            },
+        )
+        Text(
+            text = "Expires ${expiry.mediumFormat}",
+            textAlign = TextAlign.End,
+            color = MaterialTheme.colorScheme.secondary,
+            modifier = Modifier.weight(1f),
+        )
+    }
+}
+
+private val LocalDate.mediumFormat
+    get() = format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM))
+
+
+class PreviewStatusProvider: PreviewParameterProvider<DomainStatus> {
+    override val values
+        get() = DomainStatus.values().asSequence()
+}
```

Comment:
What a nice way to add params to previews! 😮 👏

Reply:
I was very pleased with this as well while coding it. 😄  Especially since the status enums are not yet finalized, afaik, so any changes needed there will automatically propagate to these previews.